### PR TITLE
fix #263: use filter in lieu of filter_stream as recommended optimization

### LINKED ISSUE #263: change filter_stream to filter
I wrote some other plugins and alway used filter, so wanted to check which is best and https://docs.fluentd.org/filter#filter-chain-optimization says to not use filter_stream ... is there any reason to keep it or can it be changed as long as the tests pass ?

### PR DESCRIPTION
**Pull request by jcantrill**
fix #263: use filter in lieu of filter_stream as recommended optimiza

### CODE REVIEW

**Comment by jcantrill** [comment]
cc @ewolinetz @alanconway @vimalk78  are takers for review?

**Review by alanconway — COMMENTED** [review]
The change looks OK, but I dont' see the changes to callers of filter_stream

**Comment by jcantrill** [comment]
> The change looks OK, but I dont' see the changes to callers of filter_stream

@alanconway I believe the callers are unknown to most users as it handled internally by fluentd when you define use of the plugin via configuration.  It is fluentd that makes the optimization.

File: test/plugin/test_filter_kubernetes_metadata.rb [hunk 1]
```diff
@@ -145,18 +145,9 @@ def emit_with_tag(tag, msg = {}, config = '
...
-      # not certain how this is possible but adding test to properly
-      # guard against this condition we have seen - test for nil,
-      # empty, no empty method, not an event stream
-      plugin = create_driver.instance
-      plugin.filter_stream('tag', nil)
-      plugin.filter_stream('tag', Fluent::MultiEventStream.new)
+      d.filtered_records
     end
 
     test 'inability to connect to the api server handles exception and doensnt block pipeline' do
```
Comment (syedriko) [inline test/plugin/test_filter_kubernetes_metadata.rb:153]: s/doensnt/doesn't

**Pull request merged**